rfc/rfc-0000-blueprint-match-fields.md

```diff
@@ -0,0 +1,110 @@
+# Meta
+[meta]: #meta
+- Name: Add MatchFields to top level of blueprints
+- Start Date: 2022-02-10
+- Author(s): @jwntrs
+- Status: Draft
+- RFC Pull Request:
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+The [template switching RFC](https://github.com/vmware-tanzu/cartographer/pull/75) introduced template `options` that included a `matchFields` selector. We should include this same selector at the `blueprint` level
+
+# Motivation
+[motivation]: #motivation
+
+- Allow entire `blueprints` to match an `owner` based on certain fields.
+
+- Ensure that the behaviour of all `selectors` is consistent.
+
+# What it is
+[what-it-is]: #what-it-is
+
+```yaml
+apiVersion: carto.run/v1alpha1
+kind: ClusterSupplyChain
+metadata:
+  name: supply-chain
+spec:
+  selector:
+    matchFields:                                                # <=========== add this
+      { key: "workload.spec.image", operation: exists }         # <=========== 
```

Comment:
since the selector here is a transitional label selector, I'd expect `matchFields` to operate like a traditional field selector. That is, to operate against the content of the Workload resource directly.

The equivalent `matchFields` proposal from RFC 9 in #75 is different since that operates on the templating context. That semantic difference may cause enough user confusing to negate the benefits of offering a field selector at this level.

```suggestion
      { key: "spec.image", operation: exists }                  # <=========== 
```

Reply:
> RFC 9 in https://github.com/vmware-tanzu/cartographer/pull/75 is different since that operates on the templating context

Reply:
> RFC 9 in https://github.com/vmware-tanzu/cartographer/pull/75 is different since that operates on the templating context

It doesn't operate on the templating context. It only operates on the workload at the moment. The syntax included the `workload` prefix to leave room for other things like params. However, since the direction from the [RFC for adding params to matchFields](https://github.com/vmware-tanzu/cartographer/pull/618) is moving towards having a top level matchParams selector, perhaps we should amend [RFC9](https://github.com/vmware-tanzu/cartographer/pull/75) to remove the workload prefix.  @emmjohnson @martyspiewak do you have thoughts on this?

Reply:
I think I'd be in favor of removing `workload` from the bottom level selectors (i.e. amending/superseding RFC9)

Reply:
removed workload from top level selector syntax